### PR TITLE
Optimize repository lookup queries to meet the expected maximums

### PR DESCRIPTION
**Pull request by josephschorr**
We were accidentally looking up more data that strictly allowed

Adds some additional assertions and testing as well

Fixes https://issues.redhat.com/browse/PROJQUAY-439

### CODE REVIEW

**Comment by josephschorr** [comment]
@dongboyan77 Mind doing a QE pass?

**Comment by josephschorr** [comment]
@dongboyan77 Actually, looks like I need to make more changes for this. So hold off.

File: data/model/oci/tag.py [hunk 1]
```diff
@@ -191,6 +191,7 @@ def get_most_recent_tag_lifetime_start(repository_ids):
     specified repository or None if none.
     """
     assert len(repository_ids) > 0 and None not in repository_ids
+    assert len(repository_ids) <= 100
```
Comment (alecmerdler) [inline data/model/oci/tag.py]: Interesting approach. It's impossible to know this requirement just from the function caller's perspective of the function signature. Is that a normal pattern?

**Comment by dongboyan77** [comment]
@josephschorr OK, I will verify it when you finished

File: data/model/oci/tag.py (hunk 1 above)
Comment (josephschorr) [reply data/model/oci/tag.py]: It is documented in the interface that calls this, but yeah, in Python there isn't really a good way to know.

**Comment by josephschorr** [comment]
@dongboyan77 Please verify when you have a chance.

**Comment by dongboyan77** [comment]
@josephschorr OK

**Comment by dongboyan77** [comment]
@josephschorr hi, QE complete verification:
1. Make sure paginated search works for them
Actual results: By default config, show 10 repos per page
2. Make sure autocompletion works for them
Screenshot from 2020-03-30 18-50-20

**Comment by dongboyan77** [comment]
Found one issue: there is no Grid View in repo page
Screenshot from 2020-03-30 18-53-49

**Comment by josephschorr** [comment]
@dongboyan77 This change removes the grid view entirely, as it cannot be paginated easily

**Comment by dongboyan77** [comment]
@josephschorr OK, got it

**Pull request merged**